src/battle_game/core/unit.cpp

```diff
@@ -1,127 +1,130 @@
-#include "battle_game/core/unit.h"
-
-#include "battle_game/core/game_core.h"
-
-namespace battle_game {
-
-namespace {
-uint32_t life_bar_model_index = 0xffffffffu;
-}  // namespace
-
-Unit::Unit(GameCore *game_core, uint32_t id, uint32_t player_id)
-    : Object(game_core, id) {
-  player_id_ = player_id;
-  lifebar_offset_ = {0.0f, 1.0f};
-  background_lifebar_color_ = {1.0f, 0.0f, 0.0f, 0.9f};
-  front_lifebar_color_ = {0.0f, 1.0f, 0.0f, 0.9f};
-  fadeout_lifebar_color_ = {1.0f, 1.0f, 1.0f, 0.5f};
-  fadeout_health_ = 1;
-  if (!~life_bar_model_index) {
-    auto mgr = AssetsManager::GetInstance();
-    life_bar_model_index = mgr->RegisterModel(
-        {{{-0.5f, 0.08f}, {0.0f, 0.0f}, {1.0f, 1.0f, 1.0f, 1.0f}},
-         {{-0.5f, -0.08f}, {0.0f, 0.0f}, {1.0f, 1.0f, 1.0f, 1.0f}},
-         {{0.5f, 0.08f}, {0.0f, 0.0f}, {1.0f, 1.0f, 1.0f, 1.0f}},
-         {{0.5f, -0.08f}, {0.0f, 0.0f}, {1.0f, 1.0f, 1.0f, 1.0f}}},
-        {0, 1, 2, 1, 2, 3});
-  }
-}
-
-void Unit::SetPosition(glm::vec2 position) {
-  position_ = position;
-}
-
-void Unit::SetRotation(float rotation) {
-  rotation_ = rotation;
-}
-
-float Unit::GetSpeedScale() const {
-  return 1.0f;
-}
-
-float Unit::GetDamageScale() const {
-  return 1.0f;
-}
-
-float Unit::BasicMaxHealth() const {
-  return 100.0f;
-}
-
-float Unit::GetHealthScale() const {
-  return 1.0f;
-}
-
-void Unit::SetLifeBarLength(float new_length) {
-  lifebar_length_ = std::min(new_length, 0.0f);
-}
-void Unit::SetLifeBarOffset(glm::vec2 new_offset) {
-  lifebar_offset_ = new_offset;
-}
-void Unit::SetLifeBarFrontColor(glm::vec4 new_color) {
-  front_lifebar_color_ = new_color;
-}
-void Unit::SetLifeBarBackgroundColor(glm::vec4 new_color) {
-  background_lifebar_color_ = new_color;
-}
-void Unit::SetLifeBarFadeoutColor(glm::vec4 new_color) {
-  fadeout_lifebar_color_ = new_color;
-}
-float Unit::GetLifeBarLength() {
-  return lifebar_length_;
-}
-glm::vec2 Unit::GetLifeBarOffset() {
-  return lifebar_offset_;
-}
-glm::vec4 Unit::GetLifeBarFrontColor() {
-  return front_lifebar_color_;
-}
-glm::vec4 Unit::GetLifeBarBackgroundColor() {
-  return background_lifebar_color_;
-}
-glm::vec4 Unit::GetLifeBarFadeoutColor() {
-  return fadeout_lifebar_color_;
-}
-
-void Unit::ShowLifeBar() {
-  lifebar_display_ = true;
-}
-void Unit::HideLifeBar() {
-  lifebar_display_ = false;
-}
-
```

Comment:
This file and `unit.h` have lots of formatter-related changes, so it should not be merged directly, but it may not affect the "approve" and score.

Reply:
Ok, I will try to make some corrections later！